From 04f56c3a2aea7e2f73a04f62788cd3dff814a631 Mon Sep 17 00:00:00 2001
From: Ionut Nechita <ionut_n2001@yahoo.com>
Date: Tue, 12 Apr 2022 22:23:51 +0300
Subject: [PATCH 53/58] SUNLIGHT: ACPI: PM: Avoid cache flush on entering S4

Description:
 - According to the ACPI spec v6.4, section 16.2 the cache flushing
required on entering to S1, S2, and S3.
 - No need to flush caches on hibernation (S4).

Bug: N/A
Change-Id: I7ae40e7330441d503a63d4fa1ba9b319624b18ae
Signed-off-by: Ionut Nechita <ionut_n2001@yahoo.com>
Signed-off-by: Kirill A. Shutemov <kirill.shutemov@linux.intel.com>
---
 drivers/acpi/sleep.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/acpi/sleep.c b/drivers/acpi/sleep.c
index c13df3f4e5fa..01afbc528370 100644
--- a/drivers/acpi/sleep.c
+++ b/drivers/acpi/sleep.c
@@ -900,8 +900,6 @@ static int acpi_hibernation_enter(void)
 {
 	acpi_status status = AE_OK;
 
-	ACPI_FLUSH_CPU_CACHE();
-
 	/* This shouldn't return.  If it returns, we have a problem */
 	status = acpi_enter_sleep_state(ACPI_STATE_S4);
 	/* Reprogram control registers */
-- 
2.35.1

